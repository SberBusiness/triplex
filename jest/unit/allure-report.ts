import {JasmineAllureReporter} from 'allure-jasmine';
import {Status, TestResult} from 'allure-js-commons';

const reporter = new JasmineAllureReporter({
    resultsDir: 'tests/out/unit/allure/allure-results',
    testMapper: (result: TestResult) => {
        if (result.status == Status.SKIPPED) result.fullName = `(WAS SKIPPED) ${result.fullName}`;
        return result;
    },
});

// @ts-ignore
jasmine.getEnv().addReporter(reporter);

export const allure = reporter.getInterface();
