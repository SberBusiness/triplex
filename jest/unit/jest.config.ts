import type {Config} from '@jest/types';

// Конфигурация Jest для запуска unit тесов.
const config: Config.InitialOptions = {
  preset: 'ts-jest',
  rootDir: '../../',
  roots: [
    '<rootDir>/src/',
    '<rootDir>/tools/',
  ],
  setupFilesAfterEnv: [
    '<rootDir>/jest/jest.setup.ts',
    '<rootDir>/jest/unit/allure-report.ts',
  ],
  moduleDirectories: [
    'node_modules',
    'src'
  ],
  moduleNameMapper: {
    "@jest/(.*)": "<rootDir>/jest/$1.ts",
    "^@sberbusiness/triplex/(.*)": "<rootDir>/src/$1"
  },
  verbose: true,
};

export default config;