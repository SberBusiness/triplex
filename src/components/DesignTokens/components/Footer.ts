import {TDesignTokenValue, TDesignTokenValues} from '@sberbusiness/triplex/components/DesignTokens/types/DesignTokenTypes';

// Название токенов компонента Footer.
export const designTokensComponentsFooterKeys = ['Background', 'Color'] as const;
// Тип, содержащий названия токенов компонента Footer.
export type TDesignTokensComponentsFooterKeys = (typeof designTokensComponentsFooterKeys)[number];
// Тип, содержащий названия токенов компонента Footer и их значения.
export type TDesignTokensComponentsFooterValue = Record<TDesignTokensComponentsFooterKeys, TDesignTokenValue>;
// Тип, содержащий названия токенов компонента Footer и их значения в светлой и темной теме.
export type TDesignTokensComponentsFooterValues = Record<TDesignTokensComponentsFooterKeys, TDesignTokenValues>;
// Тип локальных токенов компонента Footer.
export type TDesignTokensComponentsFooter = {Footer: TDesignTokensComponentsFooterValue};

// Токены компонента Footer в светлой и темной темах.
export const Footer_Tokens: TDesignTokensComponentsFooterValues = {
    Background: [{ref: 'Neutral.100'}, {ref: 'Basic.700'}], // var(--triplex-Footer-Background)
    Color: [{ref: 'Basic.700'}, {ref: 'Neutral.100'}], // var(--triplex-Footer-Color)
};
