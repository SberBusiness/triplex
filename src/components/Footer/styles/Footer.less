@import 'src/styles/style';

.footer {
    padding: @ufs-theme-page-padding-y @ufs-theme-page-padding-x;
    background: @color-asphalt-05;

    &.sticky {
        z-index: @page-header-sticky-z-index;
        position: sticky;
        bottom: 0;
    }
}

.footerDescription {
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.footerDescriptionContent {
    flex-grow: 1;
    margin-right: auto;
    .b1-text-main(@color-graphite-01);
}

.footerDescriptionControls {
    margin-left: auto;
    white-space: nowrap;
    padding-left: @ufs-theme-padding-before-controls-column;
}

// На мобильных устройствах блок с кнопками располагается строкой ниже.
@media (max-width: @screen-sm-max) {
    .footerDescription {
        flex-wrap: wrap;
        align-items: normal;
    }

    .footerDescriptionContent {
        width: 100%;
    }

    .footerDescriptionControls {
        padding-left: 0;
        white-space: normal;
        // Отступы, компенсирующие отступы у Button.
        margin-top: -16px;
        margin-left: -16px;

        > button,
        > .globalButtonDropdownExtended {
            margin-left: 16px;
            margin-top: 16px;
        }
    }

    .footerDescriptionContent + .footerDescriptionControls {
        margin-top: 0;
    }

    .footer {
        padding: @page-padding-sm;
    }
}

each(@scroll-sizes, {
    @scroll-width: @value;

    html.scroll-@{scroll-width} {
        /*
          Breakpoint для экранов менее 1024px.
          Футер при этом разрешении в лайтбоксе не фиксируется.
        */
        @media (max-width: (@lightBox-media-point-0 + @scroll-width)) {
            .globalLightBoxContent .footer,
            .globalLightBoxSideOverlay .footer {
                &.sticky {
                    position: relative;
                }
            }
        }
    }
});
