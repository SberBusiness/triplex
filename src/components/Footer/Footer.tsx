import React from 'react';
import {FooterDescription} from '@sberbusiness/triplex/components/Footer/components/FooterDescription';
import {classnames} from '@sberbusiness/triplex/utils/classnames/classnames';

/** Свойства компонента Footer. */
export interface IFooterProps extends React.HTMLAttributes<HTMLDivElement> {
    /**
     * Footer прилипает к нижней границе экрана при скролле.
     * Свойство работает на экранах высотой более 800px и шириной более 1024px.
     * */
    sticky?: boolean;
}

interface IFooterFC extends React.FC<IFooterProps> {
    Description: typeof FooterDescription;
}

/** Подвал. */
export const Footer: IFooterFC = ({children, className, sticky, ...htmlDivAttributes}) => (
    <div
        className={classnames('cssClass[footer]', className, {
            'cssClass[sticky]': Boolean(sticky),
        })}
        {...htmlDivAttributes}
        data-tx={process.env.npm_package_version}
    >
        {children}
    </div>
);

Footer.Description = FooterDescription;
Footer.displayName = 'Footer';
