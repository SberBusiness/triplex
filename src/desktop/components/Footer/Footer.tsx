import * as React from 'react';
import {FooterDescription} from '@sberbusiness/triplex/desktop/components/Footer/components/FooterDescription';
import {classnames} from '@sberbusiness/triplex/common/utils/classnames/classnames';

export interface IFooterProps extends React.HTMLAttributes<HTMLDivElement> {
    sticky?: boolean;
}

interface IFooterFC extends React.FC<IFooterProps> {
    Description: typeof FooterDescription;
}

/** Подвал. */
export const Footer: IFooterFC = ({children, className, sticky, ...htmlDivAttributes}) => (
    <div
        className={classnames('cssClass[footer]', className, {
            'cssClass[sticky]': Boolean(sticky),
        })}
        {...htmlDivAttributes}
    >
        {children}
    </div>
);

Footer.Description = FooterDescription;
Footer.displayName = 'Footer';
