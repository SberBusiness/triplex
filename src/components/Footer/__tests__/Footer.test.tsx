import React from 'react';
import renderer from 'react-test-renderer';
import {Footer} from '../Footer';
import {allure} from '@jest/unit/allure-report';

describe('Footer', () => {
    beforeEach(() => {
        allure.feature('Footer');
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(
                <Footer>
                    <Footer.Description>
                        <Footer.Description.Content>Text</Footer.Description.Content>
                        <Footer.Description.Controls>Controls</Footer.Description.Controls>
                    </Footer.Description>
                </Footer>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
