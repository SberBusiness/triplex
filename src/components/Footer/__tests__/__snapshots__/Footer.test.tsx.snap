// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div
  className="cssClass[footer]"
>
  <div
    className="cssClass[footerDescription]"
  >
    <div
      className="cssClass[footerDescriptionContent]"
    >
      Text
    </div>
    <div
      className="cssClass[footerDescriptionControls]"
    >
      Controls
    </div>
  </div>
</div>
`;
